loopy/symbolic.py

```diff
@@ -1974,6 +1975,15 @@ def simplify_using_aff(kernel, expr):
             .get_inames_domain(inames)
             .project_out_except(inames, [dim_type.set]))
 
+    non_inames = deps - set(domain.get_var_dict().keys())
+    non_inames = set([name for name in set(non_inames) if name.isidentifier()])
+    if non_inames:
+        try:
+            extra_domain = isl.BasicSet("{[" + ",".join(non_inames) + "]}")
+            domain = domain.product(extra_domain)
+        except isl.Error:
```

Comment:
What errors does this guard against?

Reply:
There were dependencies like `lid(0)`, but this error should no longer happen because I've filtered out any variables that are not valid identifiers.

Reply:
Removed in 038861f65ea2a3b201f610222a39a2e10548b1e2